### ISSUE TITLE
`phydms_prepalignment` protein dictionary key error

### ISSUE THREAD

**Issue by skhilton** [jbloomlab/phydms#25]
`phydms_prepalignment` is throwing an error when it tries to look up certain sequences in a dictionary of protein sequences during the `mafft` protein alignment. 

Here is the error message and I've attached the files to recreate the error. 
```
Beginning execution of phydms_prepalignment in directory /home/shilton/phydms_prepalignment_error at time Tue Jan  7 12:52:01 2020

Version information:
	Time and date: Tue Jan  7 12:52:01 2020
	Platform: Linux-3.13.0-143-generic-x86_64-with-debian-jessie-sid
	Python version: 3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34)  [GCC 7.3.0]
	phydms version: 2.3.1
	Bio version: 1.73
	cython version: 0.25.2
	numpy version: 1.15.0
	scipy version: 1.2.1
	matplotlib version: 3.0.3
	natsort version: 5.2.0
	sympy version: 1.0
	six version: 1.11.0
	pandas version: 0.23.4
	pyvolve version: 0.8.4
	statsmodels version: 0.8.0
	weblogolib version: 3.5.0
	PyPDF2 version: 1.26.0

Parsed the following command-line arguments:
	inseqs = bird.fasta
	alignment = out.fasta
	refseq = CY101265
	prealigned = False
	mafft = mafft
	minidentity = 0.7
	minuniqueness = 2
	purgeseqs = None
	keepseqs = None

Read 8 sequences from bird.fasta
Retained 8 after removing those specified for purging by '--purgeseqs.'
Retained 8 sequences after removing any with length not multiple of 3.
Retained 8 sequences after purging any with ambiguous nucleotides.
Retained 8 sequences after purging any with premature stops or that are otherwise un-translateable.
Using the following as reference sequence: gb:CY101265:13-2292|Organism:Influenza A virus (A/ruddy turnstone/Delaware Bay/136/1999(H3N2))|Strain Name:A/ruddy turnstone/Delaware Bay/136/1999|Protein Name:PB2 Polymerase (basic) protein 2|Gene Symbol:PB2|Segment: 1|Subtype:H3N2|Host:Ruddy Turnstone
Purged sequences encoding redundant proteins, being sure to retain reference sequence and any specified by '--keepseqs'. Overall, 7 sequences remain.
Aligning sequences with MAFFT using mafft b'v7.313 (2017/Nov/15)'
Traceback (most recent call last):
  File "/home/shilton/miniconda3/bin/phydms_prepalignment", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/shilton/phydms/scripts/phydms_prepalignment", line 354, in <module>
    main() # run the script
  File "/home/shilton/phydms/scripts/phydms_prepalignment", line 265, in main
    alignment = MAFFT_CDSandProtAlignments(seqs_unique + seqs_alwayskeep, args['mafft'], args['alignment'])
  File "/home/shilton/phydms/scripts/phydms_prepalignment", line 67, in MAFFT_CDSandProtAlignments
    prot = prot_d[head]
KeyError: 'gb_CY145192_16-2295|Organism_Influenza_A_virus__A/ruddy_turnstone/New_Jersey/Sg-00519/2008_H3N2__|Strain_Name_A/ruddy_turnstone/New_Jersey/Sg-00519/2008|Protein_Name_PB2_Polymerase__basic__protein_2|Gene_Symbol_PB2|Segment__1|Subtype_H3N2|Host_Ruddy_Turn'
```

Bernadetta found this error when she was trying to align PB2 sequences. I did some initial troubleshooting and found that the error was thrown when some sequences were included but not when all sequences were included (we could manually remove the "bad" sequences). I think that this might have to do with replacing characters in the header but I am not sure. 

[phydms_prepalignment_error.zip](https://github.com/jbloomlab/phydms/files/4032439/phydms_prepalignment_error.zip)

**Comment by jbloom** [comment]
@skhilton: Did you want me to look into this error and fixing it?

**Comment by skhilton** [comment]
@jbloom: I am going to look into it today and I'll let you know if I have questions

**Comment by jbloom** [comment]
Fixed in pull request #28
